modules/components/org.wso2.carbon.appfactory.core/src/main/java/org/wso2/carbon/appfactory/core/registry/AppFacRemoteRegistryAccessService.java

```diff
@@ -531,8 +545,135 @@ private AppFactoryResource convertResourceObject( AppFacResource appFacResource)
+            //the value is encoding base64 
+            byte[] encodedValue = Base64.encodeBase64(value.getBytes(Charset.forName("UTF-8")));
+            properties.put(key, new String(encodedValue, Charset.forName("UTF-8")));
+        } else {
+            runtimeProperty.setPropertyType(RuntimeProperty.PropertyType.ENVIRONMENT);
+            properties.put(key, value);
+        }
+
+        runtimeProperty.setName(key);
+        runtimeProperty.setProperties(properties);
+        List<RuntimeProperty> runtimeProperties = new ArrayList<RuntimeProperty>();
+        runtimeProperties.add(runtimeProperty);
+
+        ApplicationContext applicationContext = new ApplicationContext();
+        applicationContext.setCurrentStage(stage);
+        applicationContext.setId(applicationKey);
+        applicationContext.setVersion(version);
+
+        int tenantId = PrivilegedCarbonContext.getCurrentContext().getTenantId();
+        TenantInfo tenantInfo = new TenantInfo();
+        tenantInfo.setTenantId(tenantId);
+        tenantInfo.setTenantDomain(tenantDomain);
+        applicationContext.setTenantInfo(tenantInfo);
+
+        KubernetesRuntimeProvisioningService kubernetesRuntimeProvisioningService
+                = new KubernetesRuntimeProvisioningService(applicationContext);
+
+        try {
+            DeploymentConfig deploymentConfig = JDBCResourceDAO.getInstance().getDeploymentConfig(applicationKey, stage);
+            kubernetesRuntimeProvisioningService.setRuntimeProperties(runtimeProperties, deploymentConfig);
+        } catch (RuntimeProvisioningException e) {
+            String message = "Unable to create runtime property for application : " + applicationKey +
+                    " and version : " + version + " in stage : " + stage + " with property : " + key +
+                    " in tenant domain : " + tenantDomain;
+            log.error(message, e);
+            throw new AppFactoryException(message, e);
+        }
+
+    }
+
+    /**
+     * Get runtime propeties from kubernetes for application
+     *
+     * @param applicationKey id of the application
+     * @param stage          stage of the application
+     * @return list of resources as runtime properties
+     */
+    @Override
+    public List<org.wso2.carbon.appfactory.core.dto.Resource> getRuntimeProperties(String applicationKey,
+            String stage, String version) throws AppFactoryException{
+        String tenantDomain = PrivilegedCarbonContext.getCurrentContext().getTenantDomain();
+
+        if (log.isDebugEnabled()) {
+            log.debug("Getting runtime propertie for application : " + applicationKey + " and version : " + version
+                    + " in stage " + stage + " in tenant domain : " + tenantDomain);
+        }
+
+        ApplicationContext applicationContext = new ApplicationContext();
+        applicationContext.setCurrentStage(stage);
+        applicationContext.setId(applicationKey);
+        applicationContext.setVersion(version);
+
```

Comment:
This is fixed in following commit.
https://github.com/wso2/product-af/pull/800/commits/2468ef6d30141528a2e5f414e549da52ff652ae7